snuba/subscriptions/executor_consumer.py

```diff
@@ -438,3 +438,11 @@ def join(self, timeout: Optional[float] = None) -> None:
             self.__commit(
                 {message.partition: Position(message.offset, message.timestamp)}
             )
+
+        # Flush producer
+        if isinstance(self.__producer, KafkaProducer):
+            remaining = timeout - (time.time() - start) if timeout is not None else None
+            # TODO: This is just for testing. If it works expose a flush method on KafkaProducer
+            producer = self.__producer._KafkaProducer__producer  # type: ignore
+            messages = producer.flush(*[remaining] if remaining is not None else [])
+            logger.debug(f"{messages} executor messages pending delivery")
```

Comment:
I don't think you need to do this.
https://github.com/getsentry/arroyo/blob/main/arroyo/backends/kafka/consumer.py#L738-L740

The arroyo Producer has a close method that prevents it from accepting new messages and returns you a future that flushes the producer. Maybe we should ensure we log in arroyo whether the producer was flushed properly or not.

Reply:
Hmm, good point. Updated to wait for the future on close instead.